readthedocs/oauth/views.py

```diff
@@ -0,0 +1,557 @@
+
+        # If the installation was just created, we already synced the repositories and organization.
+        if created:
+            return
+
+        # We need to do a full sync of the repositories if members were added or removed,
+        # this is since we don't know to which repositories the members have access.
+        # GH doesn't send a member event for this.
+        if action in ("member_added", "member_removed"):
+            installation.service.sync()
+            return
+
+        # NOTE: installation_target should handle this instead?
+        # But I wasn't able to trigger neither of those events when renaming an organization.
+        # Maybe a bug?
+        # If the organization is renamed, we need to sync the repositories, so they use the new name.
+        if action == "renamed":
+            installation.service.sync()
+            return
+
+        if action == "deleted":
+            # Delete the organization only if it's not linked to any project.
+            # GH sends a repository and installation_repositories events for each repository
+            # when the organization is deleted.
+            # I didn't see GH send the deleted action for the organization event...
+            # But handle it just in case.
+            installation.delete_organization(data["organization"]["id"])
+            return
+
+        # Ignore other actions:
+        # - member_invited: We don't do anything with invited members.
+
+    def _handle_member_event(self):
+        """
+        Handle the member event.
+
+        Triggered when a user is added or removed from a repository.
+
+        See https://docs.github.com/en/webhooks/webhook-events-and-payloads#member.
+        """
+        data = self.request.data
+        action = data["action"]
+
+        installation, created = self._get_or_create_installation()
+
+        # If we didn't have the installation, all repositories were synced on creation.
+        if created:
+            return
+
+        if action in ("added", "edited", "removed"):
+            # Sync collaborators
+            installation.service.update_or_create_repositories([data["repository"]["id"]])
+            return
+
+        # NOTE: this should never happen.
+        raise ValidationError(f"Unsupported action: {action}")
+
+    def _handle_github_app_authorization_event(self):
+        """
+        Handle the github_app_authorization event.
+
+        Triggered when a user revokes the authorization of a GitHub App ("log in with GitHub" will no longer work).
+
+        .. note::
+
+           Revoking the authorization of a GitHub App does not uninstall the GitHub App,
+           it only revokes the OAuth2 token.
+
+        See https://docs.github.com/en/webhooks/webhook-events-and-payloads#github_app_authorization.
+        """
+        # A GitHub App receives this webhook by default and cannot unsubscribe from this event.
+        # We don't need to do anything here for now.
```

Comment:
We seem to be doing explicit `return` on some of these functions, but blank returns on others. We should probably standardize on explicit `return`.